src/api/entities/Asset/Fungible/TransferRestrictions/index.ts

```diff
@@ -57,39 +64,174 @@ export class TransferRestrictions extends Namespace<FungibleAsset> {
+            },
+          });
+
+          stats.push({
+            type: statType,
+            issuer,
+            claimType,
+            claimValue: true,
+          });
+          stats.push({
+            type: statType,
+            issuer,
+            claimType,
+            claimValue: false,
+          });
+
+          queries.push([
+            statistics.assetStats,
+            [stat1stKey, getStat2ndKey(context, claimType, true)],
+          ]);
+          queries.push([
+            statistics.assetStats,
+            [stat1stKey, getStat2ndKey(context, claimType, false)],
+          ]);
+        } else if (type === ClaimType.Jurisdiction) {
+          result.push({
+            type: statType,
+            value: new BigNumber(0),
+            claim: {
+              issuer,
+              claimType: ClaimType.Jurisdiction,
+              value: [],
+            },
+          });
+
+          const countryCodes = Object.values(CountryCode);
+
+          countryCodes.forEach(countryCode => {
+            stats.push({
+              type: statType,
+              issuer,
+              claimType: ClaimType.Jurisdiction,
+              claimValue: countryCode,
+            });
+          });
+
+          stats.push({
+            type: statType,
+            issuer,
+            claimType: ClaimType.Jurisdiction,
+            claimValue: null,
+          });
+
+          countryCodes.forEach(countryCode => {
+            queries.push([
+              statistics.assetStats,
+              [stat1stKey, getStat2ndKey(context, ClaimType.Jurisdiction, countryCode)],
+            ]);
+          });
+
+          queries.push([
+            statistics.assetStats,
+            [stat1stKey, getStat2ndKey(context, ClaimType.Jurisdiction)],
+          ]);
+        }
+      } else {
+        stats.push({
+          type: statType,
+        });
+
+        queries.push([statistics.assetStats, [stat1stKey, getStat2ndKey(context)]]);
+      }
     });
 
     const values = await requestMulti(context, queries);
 
-    return restrictions.map((restriction, index) => ({
-      restriction,
-      value: u128ToBigNumber(values[index]),
-    }));
+    const statsWithValue = stats
+      .map((stat, index) => {
+        return {
+          ...stat,
+          value:
+            stat.type === StatType.Count || stat.type === StatType.ScopedCount
+              ? u128ToBigNumber(values[index])
+              : u128ToBigNumber(values[index]).shiftedBy(-6),
+        };
+      })
+      .filter(stat => stat.value.gt(0));
+
+    statsWithValue.forEach(stat => {
+      if (!stat.claimType) {
+        result.push({
+          type: stat.type,
+          value: new BigNumber(0),
+        });
+      } else {
+        const statResult = result
+          .filter(resultStat => resultStat.claim)
+          .find(
+            resultStat =>
+              resultStat.type === stat.type &&
+              resultStat.claim!.issuer!.did === stat.issuer!.did &&
```

Comment:
I don't like the non-null assertions here.

The filter should use a type guard or this should b`?.` type check imo. The non-null is opting out of type safety.